mplc/experiment.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+"""
+An Experiment regroups multiple scenarios and enables to run them and analyze their results.
+"""
+
+import os
+import datetime
+import matplotlib.pyplot as plt
+from loguru import logger
+from pathlib import Path
+
+from . import scenario as scenario_module
+from . import utils, constants
+
+
+class Experiment:
+    def __init__(
+            self,
+            experiment_name=None,
+            scenarios_list=None,
+            nb_repeats=1,
+    ):
+        """
+        :param experiment_name: string, name of the experiment
+        :param scenarios_list: [Scenario], lists instances of the Scenario object
+        :param nb_repeats: int, number of repeats of the experiments (as an experiment includes
+                           a number of non-deterministic phenomena. Example: 5
+        """
+
+        self.name = experiment_name
+        self.scenarios_list = scenarios_list
```

Comment:
What if someone does not give any scenario list ? 
The self.scenario_list will be initialized to `None`, and the `.add_scenario()` will crash with an error like None as no method append, or sort of. It is not really user friendly to debug. I suggest that if no scenario are passed to the Experiment(), the scenario_list should be initialize to []

Reply:
outdated